### ISSUE TITLE
plugin doesn't work with NX

### ISSUE THREAD

**Issue by kosiakMD** [darraghoriordan/eslint-plugin-nestjs-typed#142]
`"@darraghor/eslint-plugin-nestjs-typed": "^4.4.0",`

either files not found or are ignored - blocks other rules to go forward even with warn

<img width="737" alt="Screenshot 2024-01-10 at 15 31 33" src="https://github.com/darraghoriordan/eslint-plugin-nestjs-typed/assets/4552929/bbcc9882-ff77-4fa8-b25a-5f3a4d065b4b">

<img width="726" alt="Screenshot 2024-01-10 at 15 31 49" src="https://github.com/darraghoriordan/eslint-plugin-nestjs-typed/assets/4552929/0ae6f656-3faa-4c61-bc5a-d13b8ca3bd1d">

ALSO:

I have next rule:
```typescript
'@darraghor/nestjs-typed/injectable-should-be-provided': [
  'error',
  {
    src: ['src/**/*.ts'],
    filterFromPaths: ['node_modules', '.test.', '.spec.', 'jest.config.ts', 'tsconfig.spec.json', 'tsconfig.scripts.json'],
  },
],
```

and got next error
```
> nx run api:lint


Linting "api"...

 >  NX   Error while loading rule '@darraghor/nestjs-typed/injectable-should-be-provided': No files matching 'src/**/*.ts' were found.

   Occurred while linting /..****../..[projectRootfolder]../apps/api/jest.config.ts
   Pass --verbose to see the stacktrace.
```

--verbose add stacktrace, here is first lines:
```
Error: Error while loading rule '@darraghor/nestjs-typed/injectable-should-be-provided': All files matched by '**/*.ts' are ignored.
Occurred while linting /..****../..[projectRootfolder]../apps/api/jest.config.ts
    at FileEnumerator.iterateFiles (/..****../..[projectRootfolder]../node_modules/eslint/lib/cli-engine/file-enumerator.js:326:27)
```

**Comment by roderik** [comment]
I struggled too, but if you define a full path for src in the local folder, it works fine. I manage our monorepo libs folder with a script to update everything everywhere https://gist.github.com/roderik/c83efef5412b7bbd7d76329c02a28338 (specific to us, do not just copy it, but the logic to get the path and update the files is usable

**Comment by darraghoriordan** [comment]
i don't use NX sorry. If get a chance to investigate it would help all nx users. you should able to turn off this rule with '@darraghor/nestjs-typed/injectable-should-be-provided': "off"

**Comment by ivan-shaban** [comment]
So as i understand we can add next:
```
{
  "extends": ["../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["**/*.ts"],
      "extends": ["plugin:@darraghor/nestjs-typed/recommended"],
      "plugins": ["@darraghor/nestjs-typed"],
      "rules": {
        "@darraghor/nestjs-typed/injectable-should-be-provided": [
          "error",
          {
            "src": ["apps/server/src/**/*.ts"],
            "filterFromPaths": ["node_modules"]
          }
        ]
      },
      "parserOptions": {
        "project": ["apps/server/tsconfig.*?.json"]
      }
    },
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
```

Where `apps/server` path to your module.

**Comment by darraghoriordan** [comment]
> Big thanks to @roderik, so as i understand we can add next:
> 
> ```
> {
>   "extends": ["../../.eslintrc.json"],
>   "ignorePatterns": ["!**/*"],
>   "overrides": [
>     {
>       "files": ["**/*.ts"],
>       "extends": ["plugin:@darraghor/nestjs-typed/recommended"],
>       "plugins": ["@darraghor/nestjs-typed"],
>       "rules": {
>         "@darraghor/nestjs-typed/injectable-should-be-provided": [
>           "error",
>           {
>             "src": ["apps/server/src/**/*.ts"],
>             "filterFromPaths": ["node_modules"]
>           }
>         ]
>       },
>       "parserOptions": {
>         "project": ["apps/server/tsconfig.*?.json"]
>       }
>     },
>     {
>       "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
>       "rules": {}
>     },
>     {
>       "files": ["*.ts", "*.tsx"],
>       "rules": {}
>     },
>     {
>       "files": ["*.js", "*.jsx"],
>       "rules": {}
>     }
>   ]
> }
> ```
> 
> Where `apps/server` path to your module.
> 
> Also i think it could be added to basic docs, because it is not obvious.

Do you use this in a commercial settings? If so can you get some of your paid time to submit a PR to update docs? 🙏
